webserver/views/api/v1/core.py

```diff
@@ -126,33 +127,41 @@ def submit_low_level_nombid():
     except ValueError as e:
         raise webserver.views.api.exceptions.APIBadRequest("Cannot parse JSON document: %s" % e)
     
-    if 'musicbrainz_recordingid' in data['metadata']['tags'].keys():
-        mbid = data['metadata']['tags']['musicbrainz_recordingid']
-        action = "mbid_submitted"
-    elif 'musicbrainz_trackid' in data['metadata']['tags'].keys():
-        mbid = data['metadata']['tags']['musicbrainz_trackid']
-        action = "mbid_submitted"
+    if 'musicbrainz_recordingid' in data['metadata']['tags'].keys() or 'musicbrainz_trackid' in data['metadata']['tags'].keys():
+        raise webserver.views.api.exceptions.APIBadRequest('### The data files contains an ID this endopoint only accept nonmbid submissions!')
     elif 'md5_encoded' in data['metadata']['audio_properties'].keys():
         md5encoded = data['metadata']['audio_properties']['md5_encoded']
         mbid = db.data.find_md5_duplicates(md5encoded)
         if mbid is not None:
             action = "md5_duplicate"
-            return jsonify({"status": "OK", "itemuuid": mbid, "action": action})
+            outputmsg = {"status": "OK", "itemuuid": mbid, "action": action}
+            return jsonify(outputmsg), 200
         else:
-            mbid = str(uuid.uuid4())
-            data['metadata']['tags']['musicbrainz_trackid'] = [mbid]
-            action = "uuidV4_generate"
+            if 'artist' in data['metadata']['tags'].keys() and 'title' in data['metadata']['tags'].keys():
+                artist = data['metadata']['tags']['artist']
+                if isinstance(artist, list):
+                    artist = artist[0]
+                title = data['metadata']['tags']['title']
+                if isinstance(title, list):
+                    title = title[0]
+                artist_data = [{'artist': artist, 'title': title}]
+                mbid = messybrainz.get_messybrainz_id(artist_data)
+                if mbid is not None:
+                    action = "messybrainz_id"
+                    try:
+                        data['metadata']['tags']['musicbrainz_trackid'] = [mbid]
+                        submit_low_level_data(mbid, data)
+                    except BadDataException as e:
+                        logging.warn(str(e))
+                        raise webserver.views.api.exceptions.APIBadRequest("%s" % e)
+                    status = "OK"
+                    outputmsg = {"status": status, "itemuuid": mbid, "action": action}
+                    return jsonify(outputmsg), 201
+                else:
```

Comment:
What is the error case that you're trying to catch here?  In HTTP, error codes are grouped based on their type. Errors that start with 2 mean "success", 3 mean "redirect", 4 mean "error caused by something the client did", and 5 "error caused by something the server did". 

I can think of these causes for an error on this endpoint, related to messybrainz data:
 * The submission doesn't have an artist or title. I would return HTTP 400 here, because the data that the user sent is bad, and it will never be accepted unless the user changes the data that they submit
 * We failed to generate a messybrainz id. This is our problem, and therefore should be a 5xx error. HTTP 503 sounds like a good option here. The idea with a 503 is that the user can probably retry with the same data later, and we'll accept it.

If we raise `NoDataFoundException` in the API then we return a status code of 404, which doesn't make sense given the two possible failure cases.